tests/filters/test_default_if_none.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+from django.template.base import VariableDoesNotExist
+
+
+@pytest.mark.parametrize(
+    "value,expected",
+    [
+        pytest.param(None, "default_value", id="none_value"),
+        pytest.param("", "", id="empty_string"),
+        pytest.param(0, "0", id="zero"),
+        pytest.param(False, "False", id="false"),
+        pytest.param("actual_value", "actual_value", id="string_value"),
+        pytest.param(42, "42", id="integer_value"),
+        pytest.param(3.14, "3.14", id="float_value"),
+        pytest.param(True, "True", id="true_value"),
+        pytest.param([], "[]", id="empty_list"),
+        pytest.param({}, "{}", id="empty_dict"),
+    ],
+)
+def test_default_if_none_with_various_values(assert_render, value, expected):
+    template = "{{ value|default_if_none:'default_value' }}"
+    assert_render(template=template, context={"value": value}, expected=expected)
+
+
+def test_default_if_none_with_missing_variable(assert_render):
+    template = "{{ missing|default_if_none:'default' }}"
+    assert_render(template=template, context={}, expected="")
+
+
+@pytest.mark.parametrize(
+    "template,expected",
+    [
+        pytest.param(
+            "{{ value|default_if_none:'DEFAULT'|lower }}", "default", id="chained_lower"
+        ),
+        pytest.param(
+            "{{ value|default_if_none:'default'|upper }}", "DEFAULT", id="chained_upper"
+        ),
+        pytest.param(
+            "{{ value|default_if_none:'<b>default</b>'|safe }}",
+            "<b>default</b>",
+            id="chained_safe",
+        ),
+    ],
+)
+def test_default_if_none_chained_filters(assert_render, template, expected):
+    assert_render(template=template, context={"value": None}, expected=expected)
+
+
+def test_default_if_none_preserves_html_safe_value(assert_render):
+    assert_render(
+        template="{{ value|safe|default_if_none:'default' }}",
+        context={"value": "<b>safe</b>"},
+        expected="<b>safe</b>",
+    )
```

Comment:
I think it would also be good to have a test where the `html` is escaped by `autoescape` or explicit `escape` or both - for both `value` and `default`.